contrib/net/http/roundtripper.go

```diff
@@ -90,3 +115,103 @@ func WrapClient(c *http.Client, opts ...RoundTripperOption) *http.Client {
 	c.Transport = WrapRoundTripper(c.Transport, opts...)
 	return c
 }
+
+type httpTraceResult struct {
+	DNSLookup        time.Duration
+	TCPConnection    time.Duration
+	TLSHandshake     time.Duration
+	ServerProcessing time.Duration
+	NameLookup       time.Duration
+	Connect          time.Duration
+	Pretransfer      time.Duration
+	StartTransfer    time.Duration
+	dnsStart         time.Time
+	dnsDone          time.Time
+	tcpStart         time.Time
+	tcpDone          time.Time
+	tlsStart         time.Time
+	tlsDone          time.Time
+	serverStart      time.Time
+	serverDone       time.Time
+	transferStart    time.Time
+	isTLS            bool
+	isReused         bool
+	remoteIP         string
+	remotePort       string
+}
+
+func withClientTrace(ctx context.Context, r *httpTraceResult) context.Context {
```

Comment:
I think that you might've lost [this comment](https://github.com/DataDog/dd-trace-go/pull/628#discussion_r402844922) in between commits.